src/dataflow-types/src/types.rs

```diff
@@ -1786,6 +1798,7 @@ pub struct IndexDesc {
     pub on_id: GlobalId,
     /// Expressions to be arranged, in order of decreasing primacy.
     pub key: Vec<MirScalarExpr>,
+    pub compute_instance_id: usize,
```

Comment:
Same as above; this appears redundant. It feels like the `DataflowBuilder` wants to stash some state, but we can probably put that on `DataflowBuilder` rather than core dataflow types.

Reply:
Ah, that makes sense. Will refactor.